Cards/CardContainer.gd

```diff
@@ -74,10 +79,23 @@ func set_active_card(card: CardBase) -> void:
 	_active_card = card
 
 
-func remove_active_card(card: CardBase) -> void:
+func finish_active_card_action(card: CardBase) -> void:
+	_discard_active_card()
 	_active_card = null
 
 
+func _discard_active_card() -> void:
+	var _card_to_discard = null
+	for card_index in _cards_queued_for_discard.size():
+		if (_cards_queued_for_discard[card_index].card_data == _active_card):
+			_card_to_discard = _cards_queued_for_discard[card_index]
```

Comment:
Maybe we could change `_active_card` to be CardWorld instead of CardBase? Might make this part cleaner

Reply:
CardWorld has a reference to its `card_data`

Reply:
Yes right, I saw that after